src/include/sof/audio/component.h

```diff
@@ -779,27 +781,35 @@ int comp_set_state(struct comp_dev *dev, int cmd);
 static inline void component_set_nearest_period_frames(struct comp_dev *current,
 						       uint32_t rate)
 {
+	uint64_t frames;
+
 	/* Sample rate is in Hz and period in microseconds.
 	 * As we don't have floats use scale divider 1000000.
 	 * Also integer round up the result.
 	 * dma buffer size should align with 32bytes which can't be
 	 * compatible with current 45K adjustment. 48K is a suitable
 	 * adjustment.
 	 */
+
 	switch (rate) {
 	case 44100:
-		current->frames = 48000 * current->period / 1000000;
+		frames = 48000;
 		break;
 	case 88200:
-		current->frames = 96000 * current->period / 1000000;
+		frames = 96000;
 		break;
 	case 176400:
-		current->frames = 192000 * current->period / 1000000;
+		frames = 192000;
 		break;
 	default:
-		current->frames = ceil_divide(rate * current->period, 1000000);
+		frames = rate;
 		break;
 	}
+
+	frames = SOF_DIV_ROUND_UP(frames * current->period, 1000000);
```

Comment:
frames = SOF_DIV_ROUND_UP(rate * current->period, 1000000);

Reply:
so this would have to be
frames = SOF_DIV_ROUND_UP((uint64_t)rate * current->period, 1000000);